src/test/miner_tests.cpp

```diff
@@ -199,6 +199,7 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
 
     LOCK(cs_main);
     fCheckpointsEnabled = false;
+    fRunTestBlockValidity = true;
```

Comment:
Instead of using the global for the tests, true can be passed in the tests directly to CreateNewBlock() as in #9858 . All other calls of CreateNewBlock() can use the global fRunTestBlockValidity